### ISSUE TITLE
[Bug]: getId() on null when no default layout is available

### ISSUE THREAD

**Issue by jensschaerer** [pimcore/pimcore#14144]
### Expected behavior

Open dataobject and show data in (first) Custom Layout available to user

### Actual behavior

Uncaught Error: Call to a member function getId() on null {"exception":"[object] (Error(code: 0): Call to a member function getId() on null at /var/www/html/vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php:516

### Steps to reproduce

I added a user with no rights defined to the user. Instead, the user has two roles: "readOnly" and "Product management". In the roles, the data object "Product" has a custom layout defined (Custom layout "readOnly" for role readOnly and "Product view" for role "Product management").

When I login as this user, I can't open data objects of type "Product". Instead I get an error 500, "Call to a member function getId() on null"

I investigated the error and it seems, that the default layout is not selected, because neither the Master layout (ID 0) exists for the user nor is any of the two assigned layouts defined as default layout:

```php
 //master layout has id 0 so we check for is_null()
            if ($currentLayoutId === null && !empty($validLayouts)) {
                if (count($validLayouts) === 1) {
                    $firstLayout = reset($validLayouts);
                    $currentLayoutId = $firstLayout->getId();
                } else {
                    foreach ($validLayouts as $checkDefaultLayout) {
                        if ($checkDefaultLayout->getDefault()) {
                            $currentLayoutId = $checkDefaultLayout->getId();
                        }
                    }
                }
            }

            if ($currentLayoutId === null && count($validLayouts) > 0) {
                $currentLayoutId = $validLayouts[0]->getId(); //<-- this is line 516 where the error occurs
            }
```

**Comment by mattamon** [comment]
Hey @jensschaerer,

I tried to reproduce the error but it seems to work for me.

Which pimcore version are you using?

Which other permissions are set up for the role?

**Comment by jensschaerer** [comment]
Hi @mattamon 
I've just tested on a new project based on Pimcore 10.5.15 and I can absolutely reproduce the error using basic conditions for easier testing:
- Created a user named "Testuser" with only permission "Objects"
- Permissions to data object class "Product"
- Added folder "ERP/Produkte" to data object workspace
- set custom layouts to "Test Read" and "Test Write" (two simple layouts for testing purposes)
- Login as "Testuser"
- Try to open object

![image](https://user-images.githubusercontent.com/1929590/214885621-c52c8baf-ea14-4a1d-9183-4cfd6da24d91.png)
![image](https://user-images.githubusercontent.com/1929590/214885868-947245bd-fa59-471f-8f5c-e715a12b9483.png)
![image](https://user-images.githubusercontent.com/1929590/214886036-6085f1bf-b644-4934-80c7-504ef68d0dd7.png)
![image](https://user-images.githubusercontent.com/1929590/214884956-6ea41a6f-dbeb-4d28-a0c0-ca144cacdc0e.png)

**Comment by mattamon** [comment]
Thank you for the clarification, I will give it another go to check if I can reproduce it too! :)

**Comment by mattamon** [comment]
Hey @jensschaerer,

I found the error. It was only happening if you had 2 custom views and none of the were defined to be the default one! 
I only tested it with one before, sorry for that! 
I just created a PR that should fix the problem.

**Comment by jensschaerer** [comment]
@mattamon thanks for your fast help, glad to see it will be in the next release 👍

**Comment by mattamon** [comment]
Resolved by #14160
